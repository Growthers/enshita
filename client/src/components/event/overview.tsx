import React, { useEffect, useState } from "react";
import { Icon } from "@iconify/react";

import type { SpeakerQuotaType } from "~/pages/event";
import EventDate from "~/components/event/date";
import { AnchorButton } from "~/components/button/anchor-button";
import { str2DateInfo } from "~/utils/date";
import styles from "~/styles/event/overview.module.scss";

type Props = {
  eventId: string;
  startDate: string;
  endDate: string;
  status:
    | "preparing"
    | "open"
    | "close"
    | "suddenOpen"
    | "suddenClose"
    | "finish";
  deadline: string;
  speakerQuotaTypeList: SpeakerQuotaType[];
  venue?: string;
};

const EventOverview: React.FC<Props> = props => {
  const {
    eventId,
    startDate,
    endDate,
    status,
    deadline,
    speakerQuotaTypeList,
    venue,
  } = props;
  const deadlineInfo = str2DateInfo(deadline);
  const [buttonText, setButtonText] = useState("");
  const [buttonLink, setButtonLink] = useState<string | undefined>(undefined);

  useEffect(() => {
    switch (status) {
      case "preparing":
        setButtonText("申し込み開始前");
        setButtonLink(undefined);
        break;
      case "open":
        setButtonText("申し込む");
        setButtonLink(`/apply/${eventId}`);
        break;
      case "close":
        setButtonText("申し込みは終了しました");
        setButtonLink(undefined);
        break;
      case "suddenOpen":
        setButtonText("飛び入りで申し込む");
        setButtonLink(`/apply/${eventId}`);
        break;
      case "suddenClose":
        setButtonText("申し込みは終了しました");
        setButtonLink(undefined);
        break;
      case "finish":
        setButtonText("イベントは終了しました");
        setButtonLink(undefined);
        break;
      default:
        setButtonText("");
        setButtonLink(undefined);
        break;
    }
  }, [eventId, status]);

  return (
    <div className={styles["enshita-event-overview"]}>
      <EventDate startStrDate={startDate} endStrDate={endDate} />
      {venue && (
        <div className={styles["enshita-event-venue"]}>
          <Icon icon="bytesize:location" />
          <p>{venue}</p>
        </div>
      )}
      <div className={styles["enshita-event-apply"]}>
        <AnchorButton
          variant="normal"
          href={buttonLink}
          boxStyles={styles["enshita-event-apply-box"]}
          isSamePage
        >
          {buttonText}
        </AnchorButton>
      </div>
      <div className={styles["enshita-event-deadline"]}>
        <time>
          {deadlineInfo
            ? `${deadlineInfo.strYear}/${deadlineInfo.strMonth}/${deadlineInfo.strDay} (${deadlineInfo.dayOfWeek}) ${deadlineInfo.strHour}:${deadlineInfo.strMinute}まで`
            : "No Date"}
        </time>
        <p>※申し込み締切時間は変更される可能性があります</p>
      </div>
      <div className={styles["enshita-event-speaker-quota-type-list"]}>
        {speakerQuotaTypeList.map(speakerQuotaType => (
          <div
            className={styles["enshita-event-speaker-quota-type"]}
            key={speakerQuotaType.id}
          >
            <p className={styles["enshita-event-speaker-quota-type-name"]}>
              {speakerQuotaType.name}
            </p>
            <p className={styles["enshita-event-speaker-quota-type-currently"]}>
              {speakerQuotaType.currentCount} / {speakerQuotaType.total}
            </p>
          </div>
        ))}
      </div>
    </div>
  );
};

EventOverview.defaultProps = {
  venue: "",
};

export default EventOverview;
