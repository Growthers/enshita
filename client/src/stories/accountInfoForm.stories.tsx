// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
