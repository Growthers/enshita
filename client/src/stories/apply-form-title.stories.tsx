import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
