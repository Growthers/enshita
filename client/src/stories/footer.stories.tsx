// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
