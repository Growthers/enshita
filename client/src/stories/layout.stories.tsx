import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",

  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = () => (
  <Layout>
    <div>hogehoge</div>
  </Layout>
);
