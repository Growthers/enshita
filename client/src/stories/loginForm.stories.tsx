// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import LoginForm from "~/components/loginForm";

type T = typeof LoginForm;
type Story = ComponentStoryObj<T>;

export default {
  title: "LoginForm",
  component: LoginForm,
} as ComponentMeta<T>;

export const Default: Story = {};
