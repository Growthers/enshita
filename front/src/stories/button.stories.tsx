import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Button from "~/components/button";

type T = typeof Button;
type Story = ComponentStoryObj<T>;

export default {
  title: "Button",
  args: { children: <div>hogehoge</div> },
  component: Button,
} as ComponentMeta<T>;

export const Default: Story = {};
