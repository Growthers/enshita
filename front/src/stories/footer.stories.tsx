import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
