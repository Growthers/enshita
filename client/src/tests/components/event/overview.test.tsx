import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/event/overview.stories";

const { Preparing, Open, Close, SuddenOpen, SuddenClose, Finish } =
  composeStories(stories);

describe("(component) EventOverview", () => {
  test("Snap Shot - preparing", () => {
    const { container } = render(<Preparing />);
    expect(container).toMatchSnapshot();
  });

  test("Snap Shot - open", () => {
    const { container } = render(<Open />);
    expect(container).toMatchSnapshot();
  });

  test("Snap Shot - close", () => {
    const { container } = render(<Close />);
    expect(container).toMatchSnapshot();
  });

  test("Snap Shot - suddenOpen", () => {
    const { container } = render(<SuddenOpen />);
    expect(container).toMatchSnapshot();
  });

  test("Snap Shot - suddenClose", () => {
    const { container } = render(<SuddenClose />);
    expect(container).toMatchSnapshot();
  });

  test("Snap Shot - finish", () => {
    const { container } = render(<Finish />);
    expect(container).toMatchSnapshot();
  });

  test("Button is Disabled - preparing", () => {
    const { getByRole } = render(<Preparing />);
    expect(getByRole("button", { name: "申し込み開始前" })).toBeDisabled();
  });

  test("Button is Enabled - open", () => {
    const { getByRole } = render(<Open />);
    expect(getByRole("button", { name: "申し込む" })).toBeEnabled();
  });

  test("Button is Disabled - close", () => {
    const { getByRole } = render(<Close />);
    expect(
      getByRole("button", { name: "申し込みは終了しました" }),
    ).toBeDisabled();
  });

  test("Button is Enabled - suddenOpen", () => {
    const { getByRole } = render(<SuddenOpen />);
    expect(getByRole("button", { name: "飛び入りで申し込む" })).toBeEnabled();
  });

  test("Button is Disabled - suddenClose", () => {
    const { getByRole } = render(<SuddenClose />);
    expect(
      getByRole("button", { name: "申し込みは終了しました" }),
    ).toBeDisabled();
  });

  test("Button is Disabled - finish", () => {
    const { getByRole } = render(<Finish />);
    expect(
      getByRole("button", { name: "イベントは終了しました" }),
    ).toBeDisabled();
  });
});
