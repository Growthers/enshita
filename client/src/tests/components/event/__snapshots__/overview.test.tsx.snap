// Vitest Snapshot v1

exports[`(component) EventOverview > Snap Shot - close 1`] = `
<div>
  <div
    class="_enshita-event-overview_11otg_1"
  >
    <div
      class="_enshita-event-date_9xck7_1"
    >
      <time>
        2022/08/27 (Sat)
      </time>
      <time>
        <span>
          19:30 ~ 23:00
        </span>
      </time>
    </div>
    <div
      class="_enshita-event-venue_11otg_10"
    >
      <span />
      <p>
        Discord
      </p>
    </div>
    <div
      class="_enshita-event-apply_11otg_21"
    >
      <button
        class="_enshita-button-box_1gcpx_17 _enshita-event-apply-box_11otg_26"
        disabled=""
        tabindex="0"
      >
        <p
          class="_enshita-button-text_1gcpx_36 undefined"
        >
          申し込みは終了しました
        </p>
      </button>
    </div>
    <div
      class="_enshita-event-deadline_11otg_32"
    >
      <time>
        2022/08/27 (Sat) 00:00まで
      </time>
      <p>
        ※申し込み締切時間は変更される可能性があります
      </p>
    </div>
    <div
      class="_enshita-event-speaker-quota-type-list_11otg_43"
    >
      <div
        class="_enshita-event-speaker-quota-type_11otg_43"
      >
        <p
          class="_enshita-event-speaker-quota-type-name_11otg_53"
        >
          3分枠
        </p>
        <p
          class="_enshita-event-speaker-quota-type-currently_11otg_57"
        >
          2
           / 
          5
        </p>
      </div>
      <div
        class="_enshita-event-speaker-quota-type_11otg_43"
      >
        <p
          class="_enshita-event-speaker-quota-type-name_11otg_53"
        >
          5分枠
        </p>
        <p
          class="_enshita-event-speaker-quota-type-currently_11otg_57"
        >
          3
           / 
          5
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`(component) EventOverview > Snap Shot - finish 1`] = `
<div>
  <div
    class="_enshita-event-overview_11otg_1"
  >
    <div
      class="_enshita-event-date_9xck7_1"
    >
      <time>
        2022/08/27 (Sat)
      </time>
      <time>
        <span>
          19:30 ~ 23:00
        </span>
      </time>
    </div>
    <div
      class="_enshita-event-venue_11otg_10"
    >
      <span />
      <p>
        Discord
      </p>
    </div>
    <div
      class="_enshita-event-apply_11otg_21"
    >
      <button
        class="_enshita-button-box_1gcpx_17 _enshita-event-apply-box_11otg_26"
        disabled=""
        tabindex="0"
      >
        <p
          class="_enshita-button-text_1gcpx_36 undefined"
        >
          イベントは終了しました
        </p>
      </button>
    </div>
    <div
      class="_enshita-event-deadline_11otg_32"
    >
      <time>
        2022/08/27 (Sat) 00:00まで
      </time>
      <p>
        ※申し込み締切時間は変更される可能性があります
      </p>
    </div>
    <div
      class="_enshita-event-speaker-quota-type-list_11otg_43"
    >
      <div
        class="_enshita-event-speaker-quota-type_11otg_43"
      >
        <p
          class="_enshita-event-speaker-quota-type-name_11otg_53"
        >
          3分枠
        </p>
        <p
          class="_enshita-event-speaker-quota-type-currently_11otg_57"
        >
          2
           / 
          5
        </p>
      </div>
      <div
        class="_enshita-event-speaker-quota-type_11otg_43"
      >
        <p
          class="_enshita-event-speaker-quota-type-name_11otg_53"
        >
          5分枠
        </p>
        <p
          class="_enshita-event-speaker-quota-type-currently_11otg_57"
        >
          3
           / 
          5
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`(component) EventOverview > Snap Shot - open 1`] = `
<div>
  <div
    class="_enshita-event-overview_11otg_1"
  >
    <div
      class="_enshita-event-date_9xck7_1"
    >
      <time>
        2022/08/27 (Sat)
      </time>
      <time>
        <span>
          19:30 ~ 23:00
        </span>
      </time>
    </div>
    <div
      class="_enshita-event-venue_11otg_10"
    >
      <span />
      <p>
        Discord
      </p>
    </div>
    <div
      class="_enshita-event-apply_11otg_21"
    >
      <a
        class="_enshita-button-box_8w723_17 _enshita-event-apply-box_11otg_26"
        href="/apply/123456789"
        rel="noopener noreferrer"
        role="button"
        target=""
      >
        <p
          class="_enshita-button-text_8w723_36 undefined"
        >
          申し込む
        </p>
      </a>
    </div>
    <div
      class="_enshita-event-deadline_11otg_32"
    >
      <time>
        2022/08/27 (Sat) 00:00まで
      </time>
      <p>
        ※申し込み締切時間は変更される可能性があります
      </p>
    </div>
    <div
      class="_enshita-event-speaker-quota-type-list_11otg_43"
    >
      <div
        class="_enshita-event-speaker-quota-type_11otg_43"
      >
        <p
          class="_enshita-event-speaker-quota-type-name_11otg_53"
        >
          3分枠
        </p>
        <p
          class="_enshita-event-speaker-quota-type-currently_11otg_57"
        >
          2
           / 
          5
        </p>
      </div>
      <div
        class="_enshita-event-speaker-quota-type_11otg_43"
      >
        <p
          class="_enshita-event-speaker-quota-type-name_11otg_53"
        >
          5分枠
        </p>
        <p
          class="_enshita-event-speaker-quota-type-currently_11otg_57"
        >
          3
           / 
          5
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`(component) EventOverview > Snap Shot - preparing 1`] = `
<div>
  <div
    class="_enshita-event-overview_11otg_1"
  >
    <div
      class="_enshita-event-date_9xck7_1"
    >
      <time>
        2022/08/27 (Sat)
      </time>
      <time>
        <span>
          19:30 ~ 23:00
        </span>
      </time>
    </div>
    <div
      class="_enshita-event-apply_11otg_21"
    >
      <button
        class="_enshita-button-box_1gcpx_17 _enshita-event-apply-box_11otg_26"
        disabled=""
        tabindex="0"
      >
        <p
          class="_enshita-button-text_1gcpx_36 undefined"
        >
          申し込み開始前
        </p>
      </button>
    </div>
    <div
      class="_enshita-event-deadline_11otg_32"
    >
      <time>
        2022/08/27 (Sat) 00:00まで
      </time>
      <p>
        ※申し込み締切時間は変更される可能性があります
      </p>
    </div>
    <div
      class="_enshita-event-speaker-quota-type-list_11otg_43"
    >
      <div
        class="_enshita-event-speaker-quota-type_11otg_43"
      >
        <p
          class="_enshita-event-speaker-quota-type-name_11otg_53"
        >
          3分枠
        </p>
        <p
          class="_enshita-event-speaker-quota-type-currently_11otg_57"
        >
          2
           / 
          5
        </p>
      </div>
      <div
        class="_enshita-event-speaker-quota-type_11otg_43"
      >
        <p
          class="_enshita-event-speaker-quota-type-name_11otg_53"
        >
          5分枠
        </p>
        <p
          class="_enshita-event-speaker-quota-type-currently_11otg_57"
        >
          3
           / 
          5
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`(component) EventOverview > Snap Shot - suddenClose 1`] = `
<div>
  <div
    class="_enshita-event-overview_11otg_1"
  >
    <div
      class="_enshita-event-date_9xck7_1"
    >
      <time>
        2022/08/27 (Sat)
      </time>
      <time>
        <span>
          19:30 ~ 23:00
        </span>
      </time>
    </div>
    <div
      class="_enshita-event-venue_11otg_10"
    >
      <span />
      <p>
        Discord
      </p>
    </div>
    <div
      class="_enshita-event-apply_11otg_21"
    >
      <button
        class="_enshita-button-box_1gcpx_17 _enshita-event-apply-box_11otg_26"
        disabled=""
        tabindex="0"
      >
        <p
          class="_enshita-button-text_1gcpx_36 undefined"
        >
          申し込みは終了しました
        </p>
      </button>
    </div>
    <div
      class="_enshita-event-deadline_11otg_32"
    >
      <time>
        2022/08/27 (Sat) 00:00まで
      </time>
      <p>
        ※申し込み締切時間は変更される可能性があります
      </p>
    </div>
    <div
      class="_enshita-event-speaker-quota-type-list_11otg_43"
    >
      <div
        class="_enshita-event-speaker-quota-type_11otg_43"
      >
        <p
          class="_enshita-event-speaker-quota-type-name_11otg_53"
        >
          3分枠
        </p>
        <p
          class="_enshita-event-speaker-quota-type-currently_11otg_57"
        >
          2
           / 
          5
        </p>
      </div>
      <div
        class="_enshita-event-speaker-quota-type_11otg_43"
      >
        <p
          class="_enshita-event-speaker-quota-type-name_11otg_53"
        >
          5分枠
        </p>
        <p
          class="_enshita-event-speaker-quota-type-currently_11otg_57"
        >
          3
           / 
          5
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`(component) EventOverview > Snap Shot - suddenOpen 1`] = `
<div>
  <div
    class="_enshita-event-overview_11otg_1"
  >
    <div
      class="_enshita-event-date_9xck7_1"
    >
      <time>
        2022/08/27 (Sat)
      </time>
      <time>
        <span>
          19:30 ~ 23:00
        </span>
      </time>
    </div>
    <div
      class="_enshita-event-venue_11otg_10"
    >
      <span />
      <p>
        Discord
      </p>
    </div>
    <div
      class="_enshita-event-apply_11otg_21"
    >
      <a
        class="_enshita-button-box_8w723_17 _enshita-event-apply-box_11otg_26"
        href="/apply/123456789"
        rel="noopener noreferrer"
        role="button"
        target=""
      >
        <p
          class="_enshita-button-text_8w723_36 undefined"
        >
          飛び入りで申し込む
        </p>
      </a>
    </div>
    <div
      class="_enshita-event-deadline_11otg_32"
    >
      <time>
        2022/08/27 (Sat) 00:00まで
      </time>
      <p>
        ※申し込み締切時間は変更される可能性があります
      </p>
    </div>
    <div
      class="_enshita-event-speaker-quota-type-list_11otg_43"
    >
      <div
        class="_enshita-event-speaker-quota-type_11otg_43"
      >
        <p
          class="_enshita-event-speaker-quota-type-name_11otg_53"
        >
          3分枠
        </p>
        <p
          class="_enshita-event-speaker-quota-type-currently_11otg_57"
        >
          2
           / 
          5
        </p>
      </div>
      <div
        class="_enshita-event-speaker-quota-type_11otg_43"
      >
        <p
          class="_enshita-event-speaker-quota-type-name_11otg_53"
        >
          5分枠
        </p>
        <p
          class="_enshita-event-speaker-quota-type-currently_11otg_57"
        >
          3
           / 
          5
        </p>
      </div>
    </div>
  </div>
</div>
`;
